test-harness/modules/runner/common.py

```diff
@@ -0,0 +1,726 @@
+    all messages to the log catcher instance.
+
+    Log messages are not formatted nor decoded in any way, thus they are
+    prefixed with syslog-protocol artifacts. This may be addressed at some
+    point though
+
+    Attributes:
+        SOCKET_PATH (str): path where the unix socket should be created
+    """
+    SOCKET_PATH = "/dev/log"
+    _log_catcher = None
+
+    def _cleanup_stale_sockets(self):
+        try:
+            os.unlink(self.SOCKET_PATH)
+        except OSError:
+            if os.path.exists(self.SOCKET_PATH):
+                raise
+
+    def _bind_socket(self):
+        """Bind to the socket specified by self.SOCKET_PATH
+
+        Worth noting is that Nginx setuids to user nobody, thus it is
+        necessary to give very open permission for the socket so that
+        it can be accessed by the AR instance
+        """
+        self._socket = socket.socket(socket.AF_UNIX, socket.SOCK_DGRAM)
+        self._socket.bind(self.SOCKET_PATH)
+        os.chmod(self.SOCKET_PATH, 0o666)
+
+    def __init__(self, log_catcher):
+        """Initialize new SyslogMock instance
+
+        Args:
+            log_catcher (object: LogCatcher()): a LogCatcher instance that is
+                going to be used by the mock to store captured messages.
+        """
+        self._log_catcher = log_catcher
+
+        self._cleanup_stale_sockets()
+        self._bind_socket()
+
+        self._log_catcher.add_fd(self._socket, log_file='syslog.stdout.log')
+
+    def stop(self):
+        """Stop the syslog mock and perform the cleanup"""
+        self._socket.close()
+        os.remove(self.SOCKET_PATH)
+
+    @property
+    def stdout_line_buffer(self):
+        """Interface for accessing log lines captured by syslog mock
+
+        Returns:
+            This returns a reference to the list that log catcher internally
+            uses to append log lines to. Thus it is *VERY* important to not to
+            modify this list and treat it as Read-Only
+        """
+        return self._log_catcher.line_buffer(self._socket)
+
+
+class LogWritter():
+    """LogWritter handles log lines gathered by LogCatcher.
+
+    It takes care of storing them internally in a list(), writing them to
+    a log file if requested, and logging the log lines to python logger
+
+    Worth noting is that this class should be embedded by LogCatcher object.
+
+    Attributes:
+        MAX_LINE_LENGTH (int): maximum length of the line that is supported
+            by all LogWritter instances.
+    """
+    MAX_LINE_LENGTH = 4096
+
+    _fd = None
+    _log_level = None
+    _log_fd = None
+    _line_buffer = None
+
+    @staticmethod
+    def _normalize_line_bytes(line_bytes):
+        """Normalize newlines in given bytes array
+
+        Args:
+            line_byes (b''): bytes array that should be normalized
+
+        Returns:
+            Normalized bytes array.
+        """
+        if len(line_bytes) >= 2 and line_bytes[-2:] == b'\r\n':
```

Comment:
Just as `str` objects, `bytes` objects have an `endswith()` method which is much nicer for this task.